kvdb/sqlite/config.go

```diff
@@ -0,0 +1,13 @@
+package sqlite
+
+import "time"
+
+// Config holds sqlite configuration data.
+//
+//nolint:lll
+type Config struct {
+	Timeout        time.Duration `long:"timeout" description:"The time after which a database query should be timed out."`
+	BusyTimeout    time.Duration `long:"busytimeout" description:"The maximum amount of time to wait for a database connection to become available for a query."`
+	MaxConnections int           `long:"maxconnections" description:"The maximum number of open connections to the database. Set to zero for unlimited."`
+	PragmaOptions  []string      `long:"pragmaoptions" description:"A list of pragma key-value pairs to set on the database. For example, 'auto_vacuum=incremental'. Note that the flag must be specified multiple times if multiple options are to be set."`
```

Comment:
Thinking in particular of users shooting themselves in the foot with options that are always dangerous.

Reply:
since this will be in the kvdb module and will have its own tag, we dont want to have to update that tag very often if not needed. So we decided to expose these options here so that it is easy to play around with them as we get more familiar with sqlite.

Reply:
Yes, I can see that. Perhaps at a later stage it will be possible to constraint certain settings for safety reasons.